bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FtpFolderWatcherHandler.java

```diff
@@ -0,0 +1,271 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+
+        if (!config.secureMode.matches("(?i)NONE|IMPLICIT|EXPLICIT")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Unsupported secure mode");
+            return;
+        }
+        currentFtpListingFile = new File(ConfigConstants.getUserDataFolder() + File.separator + "FolderWatcher"
+                + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+        try {
+            this.currentFtpListingFile = currentFtpListingFile;
+            previousFtpListing = WatcherCommon.initStorage(currentFtpListingFile);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            logger.debug("Can't write file {}, error message {}", currentFtpListingFile, e.getMessage());
+            return;
+        }
+        this.initJob = scheduler.scheduleWithFixedDelay(this::connectionKeepAlive, 0, config.pollInterval,
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> executionJob = this.executionJob;
+        ScheduledFuture<?> initJob = this.initJob;
+        if (executionJob != null) {
+            executionJob.cancel(true);
+        }
+        if (initJob != null) {
+            initJob.cancel(true);
+        }
+        if (ftp.isConnected()) {
+            try {
+                ftp.logout();
+                ftp.disconnect();
+            } catch (IOException e) {
+                logger.debug("Error terminating FTP connection: ", e);
+            }
+        }
+    }
+
+    private List<String> listDirectory(FTPClient ftpClient, String parentDir, String currentDir, boolean recursive)
+            throws IOException {
+        Instant dateNow = Instant.now();
+        List<String> dirList = new ArrayList<>();
+        final String dirToList = parentDir + ((!currentDir.equals("")) ? ("/" + currentDir) : "");
+
+        for (FTPFile file : ftpClient.listFiles(dirToList)) {
+            String currentFileName = file.getName();
+            if (currentFileName.equals(".") || currentFileName.equals("..")) {
+                continue;
+            }
+            if (file.isDirectory()) {
+                if (recursive) {
+                    try {
+                        dirList.addAll(listDirectory(ftpClient, dirToList, currentFileName, recursive));
+                    } catch (IOException e) {
+                        logger.debug("Can't read FTP directory: {}", dirToList, e);
+                    }
+                }
+            } else {
+                long diff = ChronoUnit.HOURS.between(file.getTimestamp().toInstant(), dateNow);
+
+                if (diff < config.diffHours) {
+                    file.setName(dirToList + "/" + currentFileName);
```

Comment:
I'm not too familiar with the FTP library so can you tell me why you are setting the file name here?

Reply:
not sure, probably left-overs from troubleshooting